File: CrackingTheCodingInterview/Udemy/LevelOrderTraversal2.java
public class LevelOrderTraversal2 {
    public static void main(String[] args) {
        
    }

    public List<List<Integer>> levelOrder(TreeNode root) {
        if (root == null) {
            return new ArrayList<List<Integer>>();
        }
        
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        
        while (queue.size() != 0) {
            List<Integer> currLevel = new ArrayList<>();
            int count = 0;
            int length = queue.size();

            while (count < length) {
                TreeNode currNode = queue.remove();
                currLevel.add(currNode.val);
                
                if (currNode.left != null) {
                queue.add(currNode.left);
            }
            
                if (currNode.right != null) {
                queue.add(currNode.right);
            }
                
                count++;
            }
            res.add(currLevel);
        }
        return res;
    }
}

File: CrackingTheCodingInterview/Udemy/LevelOrderTraversal2.js
if (root == null) return [];
    let res = [];
    let q = [root];
    while (q.length) {
        let count = 0;
        let len = q.length;
        let currLevel = [];
        while (count < len) {
            let currNode = q.shift();
            currLevel.push(currNode.val);
            count++;
            if (currNode.left != null) {
                q.push(currNode.left);
            }
            if (currNode.right != null) {
                q.push(currNode.right);
            }
        }
        res.push(currLevel);
    }
    return res;